### PR TITLE
Results-discriminator model and Model Calibration

### LINKED ISSUE #20: Example high confidence positive / high confidence negative sentences
This came up in meetings around this work a number of times. It would be good to produce, for each edge type, a table with high confidence positive (near 1) and high confidence negative (near 0) edges. You should definitely produce this for the discriminative model that you describe in the calibration section. It would probably also be good to produce this for the corresponding generative model.

### PR DESCRIPTION
**Pull request by danich1**
This section describes the discriminative model's performance and talks about model calibration.
Closes #20.
Added tables for confidence scores in Supplementary Figures section.

### CODE REVIEW

File: content/04.results.md
Comment (dhimmel) [inline content/04.results.md]: More description of what this means needed. Whenever discussing machine learning classifiers, let the user know what the observations are, what the features are, and what the status (outcome variable) is.

File: content/04.results.md
Comment (dhimmel) [inline content/04.results.md]: grammer problem in later half of sentence

File: content/04.results.md
Comment (dhimmel) [inline content/04.results.md]: overestimatedm

File: content/04.results.md
Comment (dhimmel) [inline content/04.results.md]: Would "0.2–0.6" be more natural of english?

File: content/04.results.md
Comment (dhimmel) [inline content/04.results.md]: diagnoal

**Review by dhimmel — COMMENTED** [review]
If these figures are being generated in another repo, I'd just link to them in that repo using a commit hash link. This way they are directly connected to their source and we don't have to host them twice

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
In the framework we are using, a generative model is trained over label functions to produce probabilistic training labels for each sentence.
Then we train a discriminative model, which has full access to a representation of the text of the sentence, to predict the generated labels.
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
The discriminative model is a convolutional neural network trained over word embeddings.
We report the results of the discriminative model using area under the precision recall curve (AUPRC) (Figure {@fig:discriminative_model_auprc}) and area under the receiver operating curve (AUROC) (Figure {@fig:discriminative_model_auroc}).    
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md:86]: Put in what the error bars represent. You should do this in all figures with error bars.

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]: Wait - isn't the disciminative model usually worse than the generative model? Am I misreading the results.

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
In the top left corner of figures {@fig:discriminative_model_auroc} and {@fig:discriminative_model_auprc} the discriminative model underperforms the generative model when only using distant supervision label functions, but the discriminative model surpasses the generative model when all label functions are included.
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]: I'm going to hold off continuing to review this section in case I'm misreading the results. Can you take a look and see if your interpretation matches mine (the discriminative model usually doesn't help, and is often worse)?

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
Even deep learning models with high precision and recall can be poorly calibrated, and the overconfidence of these models has been noted [@arxiv:1706.04599; @arxiv:1807.00263].
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
We attempted to calibrate the best performing discriminative model so that we could directly use the emitted probabilities.
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
We examined the calibration of our existing model (Figure {@fig:discriminative_model_calibration}, blue line).
We found that the DaG and CtG edge types were, though not perfectly calibrated, were somewhat aligned with the ideal calibration lines.
The CbG and GiG edges were poorly calibrated and increasing model certainty did not always lead to an increase in precision.
Applying the calibration algorithm (orange line) did not appear to bring predictions in line with the ideal calibration line, but did capture some of the uncertainty in the GiG edge type.
For this reason we use the measured precision instead of the predicted probabilities when determining how many edges could be added to existing knowledge bases with specified levels of confidence.
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

**Review by cgreene — CHANGES_REQUESTED** [review]
I made some suggestions in the calibration section. The discriminative model bit is confusing to me - are we interpreting your results differently?

**Comment by danich1** [comment]
Fixed the discriminative model description. Should make more sense now.

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
### Discriminative Model Performance

```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]: Did you already introduce these earlier in the manuscript? If so, no need to re-define them.

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
We found that the discriminative model under-performed the generative model in most cases.
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
Only for the CtD edge does the discriminative model appear to provide performance above that of the generative model, and that increased performance is only with modest numbers of label functions.
With the full set of label functions, the performance of both remains similar.
The trend observed in the generative model that one or a few mismatched label functions (off-diagonal) improves performance is retained despite the limited performance of the discriminative model.
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

Deleting a ton of this. You don't need to describe every panel of the figure (that'd take a thousand words, since an image is worth a thousand words 😉 ). You need to give the main take-home points and cite the figure for support.

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

Comment (cgreene) [reply content/04.results.md]: If this was being kept, AUROC should have been capitalized as it's an acronym. Worth checking the rest of the paper to see if you have it in caps.

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md:99]:
```suggestion
### Discriminative Model Calibration

```

File: content/04.results.md
Comment (cgreene) [inline content/04.results.md]:
```suggestion
A perfectly calibrated model would follow the dashed diagonal line.
```

File: content/90.back-matter.md
Comment (cgreene) [inline content/90.back-matter.md]:
```suggestion
### Confidence Scores for Each Edge Type 
```
What about adding a column for the uncalibrated scores too, since it's not clear that the calibration helped much?

**Review by cgreene — CHANGES_REQUESTED** [review]
Some more changes, but this is getting there!

**Comment by danich1** [comment]
@cgreene ready for review. Will fix conflict once this PR is approved.

**Review by cgreene — APPROVED** [review]
LGTM 👍

**Pull request merged**
